src/game/gameflow.c

```diff
@@ -362,27 +362,27 @@ static bool GameFlow_LoadLevelSequence(
 
         if (!strcmp(type_str, "start_game")) {
             seq->type = GFS_START_GAME;
-            seq->data = (void *)level_num;
+            seq->data = (void *)(intptr_t)level_num;
```

Comment:
Is there any purpose in using `intptr_t` instead of `uintptr_t`? I was reading about it and the one use case seemed to be for arithmetic right shift.

Reply:
It's solely to fix the warnings on Linux gcc: `cast to pointer from integer of different size [-Wint-to-pointer-cast]`.

Reply:
Ok. I just meant can we remove `(void *)`? Seems like `intptr_t` is not in pre C99 but idk if that is an issue.

Reply:
Removing the second cast to `void*` results in another error:
```
warning: assignment to ‘void *’ from ‘long int’ makes pointer from integer without a cast [-Wint-conversion]
```